modules/imgcodecs/src/loadsave.cpp

```diff
@@ -421,14 +421,11 @@ imread_( const String& filename, int flags, int hdrtype, Mat* mat=0 )
     }
 
     int scale_denom = 1;
-    if( flags > IMREAD_LOAD_GDAL )
+    if( flags > IMREAD_LOAD_GDAL && flags < IMREAD_IGNORE_ORIENTATION )
     {
-    if( flags & IMREAD_REDUCED_GRAYSCALE_2 )
-        scale_denom = 2;
-    else if( flags & IMREAD_REDUCED_GRAYSCALE_4 )
-        scale_denom = 4;
-    else if( flags & IMREAD_REDUCED_GRAYSCALE_8 )
-        scale_denom = 8;
+        /// Make it an even number as color flags are odd
+        /// We scale the image based on resize factor
+        scale_denom = ( flags % 2 == 0 ) ? flags >> 3 : ( flags - 1 ) >> 3;
```

Comment:
Not readable.
Not maintainable (after enum refactoring code may become broken).

Reply:
Thanks for the review @alalek, do I add the if conditions for greyscale and color, then it'll be readable, also I add these conditions to imread as well right?